tls/s2n_server_hello.c

```diff
@@ -165,6 +166,17 @@ static int s2n_server_hello_parse(struct s2n_connection *conn)
 
     POSIX_GUARD(s2n_server_extensions_recv(conn, in));
 
+    /**
+    *= https://tools.ietf.org/rfc/rfc8446#4.1.4
+    *# The server's extensions MUST contain "supported_versions".
+    **/
+    if (s2n_is_hello_retry_message(conn)) {
+        s2n_extension_type_id supported_versions_id;
+        POSIX_GUARD(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_SUPPORTED_VERSIONS, &supported_versions_id));
```

Comment:
Always init your variables. Assuming a method will init them for you can be dangerous.
```suggestion
        s2n_extension_type_id supported_versions_id = s2n_unsupported_extension;
        POSIX_GUARD(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_SUPPORTED_VERSIONS, &supported_versions_id));
```

Reply:
Fixed in f1cbddf